synapse/handlers/e2e_keys.py

```diff
@@ -226,16 +229,23 @@ async def do_remote_query(destination):
                 # probably be tracking their device lists. However, we haven't
                 # done an initial sync on the device list so we do it now.
                 try:
-                    if self._is_master:
-                        user_devices = await self.device_handler.device_list_updater.user_device_resync(
+                    if self.device_handler:
+                        user_device_resync = await self.device_handler.device_list_updater.user_device_resync(
                             user_id
                         )
                     else:
-                        user_devices = await self._user_device_resync_client(
+                        user_device_resync = await self._user_device_resync_client(
                             user_id=user_id
                         )
 
-                    user_devices = user_devices["devices"]
+                    if user_device_resync is None:
+                        failures[destination] = {
+                            "status": 503,
+                            "message": "Unable to resync devices",
+                        }
+                        continue
```

Comment:
It seems that this method assumes that `user_device_resync` will raise if the host cannot be reached, but it actually eats errors and returns `None` instead. 😢